### PR TITLE
Add Staff Create Event Endpoint and Tests

### PR DESCRIPTION
**Pull request by davidjnevin**

### CODE REVIEW

File: meetupselector/user/models.py [hunk 1]
```diff
@@ -53,6 +63,7 @@ class User(AbstractBaseUser, PermissionsMixin):
     )
     date_joined: models.DateTimeField = models.DateTimeField(_("date_joined"), auto_now_add=True)
     is_staff: models.BooleanField = models.BooleanField(_("is_staff"), default=False)
+    is_superuser: models.BooleanField = models.BooleanField(_("is_superuser"), default=False)
```
Comment (aalmiramolla) [inline meetupselector/user/models.py:66]: This field is not required because in `AbstrachBaseUser` you already have the superuser field.

**Review by aalmiramolla — CHANGES_REQUESTED** [review]
Please, review the comments I do!

File: meetupselector/user/models.py [hunk 2]
```diff
@@ -27,6 +27,16 @@ def create_user(self, email, password=None, **extra_fields):
...
+    def create_staffuser(self, email, password, **extra_fields):
+        """Create a superuser."""
+        extra_fields.setdefault("is_superuser", False)
+        extra_fields.setdefault("is_staff", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError("Staffuser must have is_staff=True.")
+
+        return self._create_user(email, password, **extra_fields)
+
```
Comment (aalmiramolla) [inline meetupselector/user/models.py:39]: I have doubts if this method is necessary.

File: tests/api/routes/test_events.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+from tests.utils.builders import UserBuilder
+
+
+@freeze_time("2022-10-26 23:23:23")
+@pytest.mark.django_db
+def test_is_staff_create_event(client, reverse_url):
+    url = reverse_url("event")
+    name = "eventName"
+    description = "description"
+    meetup_link = "meetup_link"
```
Comment (aalmiramolla) [inline tests/api/routes/test_events.py:26]: If you use URLField you have to put a valid URL here.

File: tests/api/routes/test_events.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    )
+
+
+@freeze_time("2022-10-26 23:23:23")
+@pytest.mark.django_db
+def test_is_not_staff_create_event(client, reverse_url):
+    url = reverse_url("event")
+    name = "eventName"
+    description = "description"
+    meetup_link = "meetup_link"
```
Comment (aalmiramolla) [inline tests/api/routes/test_events.py:86]: The same as the other test. Put a valid URL

File: meetupselector/user/models.py (hunk 2 above)
Comment (aalmiramolla) [reply meetupselector/user/models.py:39]: You can use User.objects.create_user(email, password, is_staff=True)

File: tests/utils/builders/user_builder.py [hunk 5]
```diff
@@ -31,12 +36,11 @@ def with_password(self, password: str) -> "UserBuilder":
...
-            description=self._description,
-            is_staff=self._is_staff,
-            date_joined=self._date_joined,
-        )
+        user, created = User.objects.get_or_create(email=self._email)
         user.set_password(self._password)
+        user.description = self._description
+        user.is_staff = self._is_staff
+        user.is_superuser = self._is_superuser
+        user.date_joined = self._date_joined
```
Comment (aalmiramolla) [inline tests/utils/builders/user_builder.py:44]: In a builder, you always want to create the resource. As is only used in tests, you know which users are created, so revert this change.

File: tests/utils/builders/event_builder.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth import get_user_model
+
+from meetupselector.proposals.models import Event
+
+User = get_user_model()
```
Comment (aalmiramolla) [inline tests/utils/builders/event_builder.py:5]: This is not used. Remove it.

**Review by aalmiramolla — CHANGES_REQUESTED** [review]

File: meetupselector/proposals/models.py [hunk 7]
```diff
@@ -63,3 +63,24 @@ class Meta:
...
+
+
+class Event(models.Model):
+    id = models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    name = models.CharField(max_length=255, verbose_name=_("name"))
+    description = models.TextField(verbose_name=_("description"))
+    location = models.CharField(max_length=255, verbose_name=_("location"))
+    meetup_link = models.CharField(max_length=255, verbose_name=_("meetup_link"))
```
[the remark this answers is not in the document]
Comment (davidjnevin) [reply meetupselector/proposals/models.py:75]: Ok. I will change that.

File: meetupselector/user/models.py (hunk 2 above)
Comment (davidjnevin) [reply meetupselector/user/models.py:39]: As above: I had problems creating a staff user using the User.objects.create_user(email, password, is_staff=True) method. This was a solution to create a staff member and not a supersuser.

File: tests/api/routes/test_events.py (hunk 3 above)
Comment (davidjnevin) [reply tests/api/routes/test_events.py:26]: Ok. I will change that.

File: tests/api/routes/test_events.py (hunk 4 above)
Comment (davidjnevin) [reply tests/api/routes/test_events.py:86]: Ok.

File: tests/utils/builders/user_builder.py (hunk 5 above)
Comment (davidjnevin) [reply tests/utils/builders/user_builder.py:44]: OK.

File: tests/utils/builders/event_builder.py (hunk 6 above)
Comment (davidjnevin) [reply tests/utils/builders/event_builder.py:5]: OK.

File: meetupselector/proposals/models.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (davidjnevin) [reply meetupselector/proposals/models.py:75]: Done

File: tests/api/routes/test_events.py (hunk 3 above)
Comment (davidjnevin) [reply tests/api/routes/test_events.py:26]: Done

File: tests/api/routes/test_events.py (hunk 4 above)
Comment (davidjnevin) [reply tests/api/routes/test_events.py:86]: Done

File: tests/utils/builders/user_builder.py (hunk 5 above)
Comment (davidjnevin) [reply tests/utils/builders/user_builder.py:44]: Done.

File: tests/utils/builders/event_builder.py (hunk 6 above)
Comment (davidjnevin) [reply tests/utils/builders/event_builder.py:5]: Done

**Review by aalmiramolla — APPROVED** [review]
LGTM
